src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -143,20 +144,22 @@ export const createSavedQueryService = (
     id: string;
     attributes: SerializedSavedQueryAttributes;
   }) => {
-    let queryString;
+    let parsedQuery;
```

Comment:
How come the rename?

Reply:
the parsed one could be a string or object. I am thinking queryString is a bit ambiguous.

Reply:
I mean `queryString: string | object`
since it is not necessary a string, should we rename it to avoid confusion?

Reply:
more tests and explain is here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1254#issuecomment-1061191659